src/vmm/src/devices/virtio/queue.rs

```diff
@@ -95,38 +89,20 @@ pub struct DescriptorChain<'a, M: GuestMemory = GuestMemoryMmap> {
     pub next: u16,
 }
 
-impl<'a, M: GuestMemory> DescriptorChain<'a, M> {
+impl DescriptorChain {
     /// Creates a new `DescriptorChain` from the given memory and descriptor table.
     ///
     /// Note that the desc_table and queue_size are assumed to be validated by the caller.
-    fn checked_new(
-        mem: &'a M,
-        desc_table: GuestAddress,
-        queue_size: u16,
-        index: u16,
-    ) -> Option<Self> {
-        if index >= queue_size {
+    fn checked_new(desc_table_ptr: *const Descriptor, queue_size: u16, index: u16) -> Option<Self> {
+        if queue_size <= index {
```

Comment:
:) What's with flipping the comparison?

Reply:
Just a preference. Small items on the left, big on the right.